Fireworks/extensions/database.py

```diff
@@ -2,22 +2,21 @@
 from sqlalchemy import Table, Column, Integer, MetaData
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker, Query
-from Fireworks import Message, cat
-from Fireworks.source import Source, PassThroughSource
+from Fireworks import Message, cat, Pipe
 import numpy as np
 import pandas as pd
 
-Base = declarative_base()
+#TODO: Develop a way to infer schemas from Messages and Pipes to make save/load easier.
 
-class TableSource(PassThroughSource):
+class TablePipe(Pipe):
     """
-    Represents an SQLalchemy Table while having the functionality of a Source.
+    Represents an SQLalchemy Table while having the functionality of a Pipe.
     """
-    def __init__(self, table, engine, columns = None, inputs = None, **kwargs):
+    def __init__(self, table, engine, columns = None, input = None, **kwargs):
 
-        super().__init__(inputs=inputs, **kwargs)
-        Session = sessionmaker(bind=engine)
-        self.session = Session()
+        super().__init__(input=input, **kwargs)
+        self.sessionmaker = sessionmaker(bind=engine)
+        self.reset_session()
```

Comment:
don't create a session at instance. Make it so that the use has to explicitly define when a session begins